Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -27,20 +50,38 @@ class APIClient {
         sessionConfiguration: URLSessionConfiguration,
         requestEncoder: RequestEncoder,
         requestDecoder: RequestDecoder,
-        CDNClient: CDNClient
+        CDNClient: CDNClient,
+        tokenRefresher: @escaping (ClientError, @escaping () -> Void) -> Void
```

Comment:
This closure is a bit hard to understand, wouldn't be easier to introduce a new class like `TokenRefresher` that would handle everything about the refreshing? The state, making refresh requests, etc... ? With the refreshing logic all inside the `request()` function, it makes it harder to understand, especially with all the queue's handling. With a `TokenRefresher` that would be all inside it